lib/tbot/workloadidentity/workloadattest/podman/client.go

```diff
@@ -0,0 +1,164 @@
+/*
+ * Teleport
+ * Copyright (C) 2025  Gravitational, Inc.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+// Package podman implements bindings for a small subset of Podman's REST API.
+//
+// We use this instead of importing github.com/containers/podman/v5/pkg/bindings
+// because we only use a couple of endpoints, so it's not worth taking on its
+// transitive dependencies.
+package podman
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net"
+	"net/http"
+	"net/url"
+
+	"github.com/gravitational/trace"
+)
+
+// APIVersion is the version of the Podman REST API we support.
+//
+// Note: Podman doesn't really seem to check or enforce the version segment
+// of the URL path, so setting it doesn't have much of an effect.
+//
+// https://github.com/containers/podman/blob/62cde17193e1469de15995ba78bd909fd07dffc6/pkg/api/server/handler_api.go#L69-L74
+const APIVersion = "v4.0.0"
+
+// Client for Podman's HTTP REST API.
+type Client struct{ httpClient *http.Client }
+
+// NewClient creates a Client for the API service at the given address.
+//
+// Note: addr must be in the form `unix://path/to/socket`, we do not currently
+// support connecting to the Podman service over TCP because it's generally
+// insecure.
+func NewClient(addr string) (*Client, error) {
+	u, err := url.Parse(addr)
+	if err != nil {
+		return nil, trace.Wrap(err, "invalid address: %q", addr)
+	}
+	if u.Scheme != "unix" {
+		return nil, trace.BadParameter("only unix domain sockets are supported")
+	}
+	return &Client{
+		httpClient: &http.Client{
```

Comment:
I can never recall if there's a sensible default timeout on http.Client, if not, we may wish to add one here. The other option would be to use a context.WithTimeout further up in the attestation process to control the amount of time any attestor can take. WDYT?

Reply:
Ah yes! There isn't, so I've added one now - good catch.

For now, I think the API calls are the only context deadline-honoring operations, but we can definitely revisit this in the future and add a context in the root attestor.